plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Bitwise.hs

```diff
@@ -0,0 +1,564 @@
+        ]
+  let rhsRotation = mkIterAppNoAnn (builtin () PLC.BitwiseRotate) [
+        mkConstant @ByteString () bs,
+        mkConstant @Integer () (fromIntegral j)
+        ]
+  let rhsIndex = mkIterAppNoAnn (builtin () PLC.ModInteger) [
+        mkConstant @Integer () (fromIntegral $ i + j),
+        mkConstant @Integer () (fromIntegral bitLen)
+        ]
+  let rhs = mkIterAppNoAnn (builtin () PLC.ReadBit) [
+        rhsRotation,
+        rhsIndex
+        ]
+  evaluateTheSame lhs rhs
+
+csbRotate :: Property
+csbRotate = property $ do
+  bs <- forAllByteString
+  i <- forAll . Gen.integral . Range.linear (negate 512) $ 512
+  let lhs = mkIterAppNoAnn (builtin () PLC.CountSetBits) [
+        mkConstant @ByteString () bs
+        ]
+  let rhsInner = mkIterAppNoAnn (builtin () PLC.BitwiseRotate) [
+        mkConstant @ByteString () bs,
+        mkConstant @Integer () i
+        ]
+  let rhs = mkIterAppNoAnn (builtin () PLC.CountSetBits) [
+        rhsInner
+        ]
+  let compareExp = mkIterAppNoAnn (builtin () PLC.EqualsInteger) [
+        lhs,
+        rhs
+        ]
+  evaluateAndVerify (mkConstant @Bool () True) compareExp
+
+-- Helpers
+
+forAllByteString :: PropertyT IO ByteString
+forAllByteString = forAllWith hexShow . Gen.bytes . Range.linear 0 $ 1024
+
+forAllByteString1 :: PropertyT IO ByteString
+forAllByteString1 = forAllWith hexShow . Gen.bytes . Range.linear 1 $ 1024
+
+forAllNonZeroByteString :: PropertyT IO ByteString
+forAllNonZeroByteString =
+  forAllWith hexShow . Gen.filterT (BS.any (/= 0x00)) . Gen.bytes . Range.linear 0 $ 1024
+
+evaluateAndVerify ::
+  UPLC.Term UPLC.Name UPLC.DefaultUni UPLC.DefaultFun () ->
+  PLC.Term UPLC.TyName UPLC.Name UPLC.DefaultUni UPLC.DefaultFun () ->
+  PropertyT IO ()
+evaluateAndVerify expected actual =
+  case typecheckEvaluateCek def defaultBuiltinCostModelForTesting actual of
+    Left x -> annotateShow x >> failure
+    Right (res, logs) -> case res of
+      PLC.EvaluationFailure   -> annotateShow logs >> failure
+      PLC.EvaluationSuccess r -> r === expected
+
+evaluateTheSame ::
+  PLC.Term UPLC.TyName UPLC.Name UPLC.DefaultUni UPLC.DefaultFun () ->
+  PLC.Term UPLC.TyName UPLC.Name UPLC.DefaultUni UPLC.DefaultFun () ->
+  PropertyT IO ()
+evaluateTheSame lhs rhs =
+  case typecheckEvaluateCek def defaultBuiltinCostModelForTesting lhs of
+    Left x -> annotateShow x >> failure
+    Right (resLhs, logsLhs) -> case typecheckEvaluateCek def defaultBuiltinCostModelForTesting rhs of
+      Left x -> annotateShow x >> failure
+      Right (resRhs, logsRhs) -> case (resLhs, resRhs) of
+        (PLC.EvaluationFailure, PLC.EvaluationFailure) -> do
+          annotateShow logsLhs
+          annotateShow logsRhs
+          failure
+        (PLC.EvaluationSuccess rLhs, PLC.EvaluationSuccess rRhs) -> rLhs === rRhs
+        (PLC.EvaluationFailure, _) -> annotateShow logsLhs >> failure
+        (_, PLC.EvaluationFailure) -> annotateShow logsRhs >> failure
+
+hexShow :: ByteString -> String
```

Comment:
There's another copy of this in Laws.hs.  Maybe just have one copy somewhere?

Reply:
It's probably better to have these somewhere that other helpers live, agreed.